Imported stdlib string constant cannot be concatenated to a string literal

The following program `sample.go` triggers a panic:

```go
package main

import (
        "fmt"
        "time"
)

func main() {
        fmt.Println(`"` + time.RFC3339Nano + `"`)
}
```

Expected result:
```console
$ go run ./sample.go
"2006-01-02T15:04:05.999999999Z07:00"
```

Got:
```console
$ yaegi ./sample.go
run: ./sample.go:9:14: invalid operation: mismatched types string and
```

Found in the wild here: https://github.com/Azure/go-autorest/blob/6184171f42979bea0ac05eb62163f6eb5d9e0d45/autorest/date/time.go#L26

I managed to get it to work on my local system by making the following change to extract.go (and regenerating stdlib/go1_16_time.go):

```diff
diff --git a/extract/extract.go b/extract/extract.go
index f6b59b3..f8844fe 100644
--- a/extract/extract.go
+++ b/extract/extract.go
@@ -299,9 +299,6 @@ func fixConst(name string, val constant.Value, imports map[string]bool) string {
                str string
        )
        switch val.Kind() {
-       case constant.String:
-               tok = "STRING"
-               str = val.ExactString()
        case constant.Int:
                tok = "INT"
                str = val.ExactString()
```

--- a/_test/issue-1089.go
+++ b/_test/issue-1089.go
@@ -1,0 +1,13 @@
+package main
+
+import (
+	"fmt"
+	"time"
+)
+
+func main() {
+	fmt.Println(`"` + time.RFC3339Nano + `"`)
+}
+
+// Output:
+// "2006-01-02T15:04:05.999999999Z07:00"

--- a/interp/type.go
+++ b/interp/type.go
@@ -1188,8 +1188,7 @@ func (t *itype) id() (res string) {
 }
 
 func constTypeString(t reflect.Type) (s string) {
-	e := reflect.New(t).Elem()
-	cv, ok := e.Interface().(constant.Value)
+	cv, ok := reflect.New(t).Elem().Interface().(constant.Value)
 	if !ok {
 		return
 	}

--- a/interp/type.go
+++ b/interp/type.go
@@ -1172,6 +1172,10 @@ func (t *itype) id() (res string) {
 		}
 		res += "}"
 	case valueT:
+		if isConstantValue(t.rtype) {
+			res = constTypeString(t.rtype)
+			break
+		}
 		res = ""
 		if t.rtype.PkgPath() != "" {
 			res += t.rtype.PkgPath() + "."
@@ -1181,6 +1185,27 @@ func (t *itype) id() (res string) {
 		res = "..." + t.val.id()
 	}
 	return res
+}
+
+func constTypeString(t reflect.Type) (s string) {
+	e := reflect.New(t).Elem()
+	cv, ok := e.Interface().(constant.Value)
+	if !ok {
+		return
+	}
+	switch cv.Kind() {
+	case constant.Bool:
+		s = "bool"
+	case constant.Int:
+		s = "int"
+	case constant.String:
+		s = "string"
+	case constant.Float:
+		s = "float64"
+	case constant.Complex:
+		s = "complex128"
+	}
+	return
 }
 
 // zero instantiates and return a zero value object for the given type during execution.

